### ISSUE TITLE
whitespace rule is confusing--can I enforce "no space?"

### ISSUE THREAD

**Issue by giladgray** [palantir/tslint#1588]
the [`whitespace`](http://palantir.github.io/tslint/rules/whitespace/) rule accepts up to 7 flags. is it possible to express the following desire?
- one of the flags is "true," as in i want whitespace in that case
- another flag is "false," as in i want to validate that there _is no_ whitespace in that case

(i suspect not...)

stylelint makes this really easy by defining each case as its own rule (often as two `-space-before` and `-space-after` rules, [see `function-comma-*` here](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/rules.md#function)). perhaps its worth considering breaking this rule into many smaller ones for each piece of the puzzle:

``` js
// require whitespace after
"whitespace-branch": [true, "after"],
// require no whitespace (but rule is enabled)
"whitespace-type": [true],
```

**Comment by JoshuaKGoldberg** [comment]
@giladgray coming back to this: what would you propose we do here? Is the `whitespace` rule now configurable enough?

**Comment by JoshuaKGoldberg** [comment]
Also relevant: discussions in #628 and linked issues.

**Comment by JoshuaKGoldberg** [comment]
Formatting rule issue without discussion for several months? Time to close 😄
